iceoryx_utils/include/iceoryx_utils/cxx/type_traits.hpp

```diff
@@ -21,6 +21,50 @@ namespace iox
 {
 namespace cxx
 {
+template <typename SignatureType>
+class function_ref;
+
+///
+/// @brief Identifies structs as being "optional" by the presence of an "if_empty" method.
+///
+template <typename T, typename = void>
+struct is_optional : std::false_type
+{
+};
+template <typename T>
+struct is_optional<T, std::void_t<decltype(std::declval<T>().has_value())>> : std::true_type
```

Comment:
This implementation is wrong and will be removed as soon as the open point described in the PR description is resolved.